src/DynamicRupture/Output/Base.cpp

```diff
@@ -214,77 +216,329 @@ void Base::updateElementwiseOutput() {
+
+        totalTractions(DirectionID::Strike, level, i) = rotatedTraction[3] + rotatedInitStress[3];
+        totalTractions(DirectionID::Dip, level, i) = rotatedTraction[5] + rotatedInitStress[5];
+        totalTractions(DirectionID::Normal, level, i) = local.p - local.pf + rotatedInitStress[0];
+      }
+
+      auto& ruptureVelocity = std::get<VariableID::RuptureVelocity>(outputData.vars);
+      if (ruptureVelocity.isActive) {
+        ruptureVelocity(level, i) = this->computeRuptureVelocity();
+      }
+
+      auto& peakSlipsRate = std::get<VariableID::PeakSlipRate>(outputData.vars);
+      if (peakSlipsRate.isActive) {
+        auto* peakSR = local.layer->var(drDescr->peakSlipRate);
+        peakSlipsRate(level, i) = peakSR[local.ltsId][local.nearestGpIndex];
+      }
+
+      auto& dynamicStressTime = std::get<VariableID::DynamicStressTime>(outputData.vars);
+      if (dynamicStressTime.isActive) {
+        auto* dynStressTime = (local.layer->var(drDescr->dynStressTime));
+        dynamicStressTime(level, i) = dynStressTime[local.ltsId][local.nearestGpIndex];
+      }
+
+      auto& slipVectors = std::get<VariableID::Slip>(outputData.vars);
+      if (slipVectors.isActive) {
+        VrtxCoords crossProduct = {0.0, 0.0, 0.0};
+        MeshTools::cross(strike, tangent1, crossProduct);
+
+        double cos1 = MeshTools::dot(strike, tangent1);
+        double scalarProd = MeshTools::dot(crossProduct, normal);
+
+        // Note: cos1**2 can be greater than 1.0 because of rounding errors -> min
+        double sin1 = std::sqrt(1.0 - std::min(1.0, cos1 * cos1));
+        sin1 = (scalarProd > 0) ? sin1 : -sin1;
+
+        auto* slip1 = local.layer->var(drDescr->slip1);
+        auto* slip2 = local.layer->var(drDescr->slip2);
+
+        slipVectors(DirectionID::Strike, level, i) =
+            cos1 * slip1[local.ltsId][local.nearestGpIndex] -
+            sin1 * slip2[local.ltsId][local.nearestGpIndex];
+
+        slipVectors(DirectionID::Dip, level, i) = sin1 * slip1[local.ltsId][local.nearestGpIndex] +
+                                                  cos1 * slip2[local.ltsId][local.nearestGpIndex];
+      }
+      this->outputSpecifics(outputData, level, i);
     }
 
-    auto& peakSlipsRate = std::get<VariableID::PeakSlipRate>(outputData.vars);
-    if (peakSlipsRate.isActive) {
-      peakSlipsRate(level, i) = peakSR[ltsId][nearestGaussPoint];
+    if (type == OutputType::AtPickpoint) {
+      outputData.cachedTime[outputData.currentCacheLevel] = time;
+      outputData.currentCacheLevel += 1;
     }
+  }
+}
+
+void Base::computeLocalStresses() {
+  real pFactorPlus = local.waveSpeedsPlus->pWaveVelocity * local.waveSpeedsPlus->density;
+  real pFactorMinus = local.waveSpeedsMinus->pWaveVelocity * local.waveSpeedsMinus->density;
+
+  real sFactorPlus = local.waveSpeedsPlus->sWaveVelocity * local.waveSpeedsPlus->density;
+  real sFactorMinus = local.waveSpeedsMinus->sWaveVelocity * local.waveSpeedsMinus->density;
```

Comment:
These are actually the impedances: https://github.com/SeisSol/SeisSol/blob/dr/cpp/src/Initializer/CellLocalMatrices.cpp#L518-L521 and could be reused.

Reply:
thanks. I changed it